src/executor/evaluate/error.rs

```diff
@@ -42,4 +45,7 @@ pub enum EvaluateError {
 
     #[error("unreachable empty aggregate value: {0:?}")]
     UnreachableEmptyAggregateValue(Aggregate),
+
+    #[error("out of range: {0}")]
+    OutOfRange(String),
```

Comment:
Ok, here let's simply depend on Rust compiler.
Rust compiler will return `NaN` and we can use that rather than returning explicit error.